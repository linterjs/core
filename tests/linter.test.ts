import { vol } from "memfs";
import Linter from "../src/linter";

jest.mock("fs");

afterEach(() => {
  vol.reset();
  jest.resetModules();
});

describe("Create Linter", () => {
  test("without any linter providers installed", () => {
    vol.fromJSON({
      "package.json": "{}",
    });

    expect(() => {
      const linter = new Linter();
    }).toThrowErrorMatchingSnapshot();
  });

  test("without any linter providers installed and a log level set via constructor parameter", () => {
    vol.fromJSON({
      "package.json": "{}",
    });

    expect(() => {
      const linter = new Linter({ logLevel: "debug" });
    }).toThrowErrorMatchingSnapshot();
  });

  test("with a linter provider installed", () => {
    vol.fromJSON({
      "package.json": JSON.stringify({
        devDependencies: {
          "@linter/provider-eslint": "1.0.0",
        },
      }),
    });

    const linter = new Linter();
    expect(linter).toBeInstanceOf(Linter);
  });
});
