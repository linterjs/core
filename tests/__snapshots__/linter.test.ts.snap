// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Linter without any linter providers installed 1`] = `"No linter providers found. Please install at least one linter provider package."`;

exports[`Create Linter without any linter providers installed and a log level set via constructor parameter 1`] = `"No linter providers found. Please install at least one linter provider package."`;

exports[`Create Linter without any linter providers installed and a log level set via env 1`] = `"No linter providers found. Please install at least one linter provider package."`;
